app/src/main/java/fi/aalto/legroup/achso/entities/UserPool.java

```diff
@@ -0,0 +1,55 @@
+package fi.aalto.legroup.achso.entities;
+
+import java.util.ArrayList;
+
+/**
+ * Use this class to store users memory efficiently.
+ * Maps User objects to indices that can later retrieve the actual User back.
+ */
+public class UserPool {
```

Comment:
What’s your opinion about hiding the constructor? I don’t see any instance methods.